### ISSUE TITLE
Add MonadTrans

### ISSUE THREAD

**Issue by treeowl** [andrewthad/quickcheck-classes#130]
I don't see anything for `MonadTrans`. I recommend using this base monad for testing it:

```haskell
data TestF a = TestF !Int [a]
  deriving (Functor, Eq, Show)
newtype TestM a = TestM (Free TestF a)
```

where it's easier to define a `Free` monad locally (with lots of `StandaloneDeriving`) than to do the `Show1`, `Eq1`, etc., gunk needed to use the one in `Free`.

**Comment by treeowl** [comment]
Of course, you could also unpack `TestF` into the free monad for efficiency. Dunno if it's worth the loss of clarity.

**Comment by andrewthad** [comment]
I'm fine with this if anyone wants to implement it. I'm not terribly concerned about efficiency or even about the particular implementation, as long as it works.

**Comment by treeowl** [comment]
I can try, though I'm not really familiar with the way things are done in this package. `MonadTrans` properties are really just the most basic one might want for `lift`. I have much less of a clue about how one might effectively test that it lifts `MonadReader`, `MonadError`, etc., properties correctly.
